File: ShopAppSpring/src/main/java/com/duchung/shopappspring/dtos/ExportTransactionDocumentDTO.java
package com.duchung.shopappspring.dtos;

import com.duchung.shopappspring.models.ExportTransactionDocumentDetail;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ExportTransactionDocumentDTO {
    private String name;

    @JsonProperty("user_id")
    private Long userId;

    @JsonProperty("raw_products")
    List<ExportRawProductDTO> rawProducts;
}

File: shopapp-angular/src/app/manager/dtos/export.transaction.document.dto.ts
import { ExportRawProductDTO } from "./export.raw.product.dto";

export class ExportTransactionDocumentDTO {
  name: string;
  user_id: number;
  raw_products: ExportRawProductDTO[];

  constructor(data: any) {
    this.name = data.name;
    this.user_id = data.user_id;
    this.raw_products = data.raw_products;
  }
}
